lib/xmldoc.js

```diff
@@ -25,9 +25,9 @@
    * behaves like an XmlElement by inheriting its attributes and functions.
    */
 
-  function XmlElement(tag) {
+  function XmlElement(tag, parser) {
     // Capture the parser object off of the XmlDocument delegate
-    var parser = delegates[delegates.length - 1].parser;
+    var parser = (delegates[delegates.length - 1]?.parser) ?? parser;
```

Comment:
allowing the author to inject their own parser is optional, though I thought it could be helpful. PR could be amended to remove this functionality.